### PR TITLE
docs: improve accessibility of http example 

### PR DESCRIPTION
**Pull request by kapunahelewong**
Fixes contrast issues.
Makes labels explicitly reference inputs.
Adds padding to buttons and inputs so they are an easier target for touch or pointer.
Changes label to explicitly state intent.
Changes markup to make it more semantic.
Updates CSS.
Updates "simple" wording.
Fixes li delete button, contrast, and responsiveness. 

## Before
Screen Shot 2021-03-02 at 2 58 56 PM

## After
Screen Shot 2021-03-02 at 2 39 36 PM

### CODE REVIEW

**Review by zarend — APPROVED** [review]
👍

File: aio/content/examples/http/src/app/heroes/heroes.component.css [hunk 1]
```diff
@@ -1,73 +1,66 @@
...
 
 .heroes a {
-  color: #888;
+  color: black;
   text-decoration: none;
-  position: relative;
   display: block;
-  width: 250px;
+  font-size: 1.2rem;
+  background-color: #EEE;
```
Comment (gkalpak) [inline aio/content/examples/http/src/app/heroes/heroes.component.css:28]: Nit (for consistency):

```suggestion
  background-color: #eee;
```

File: aio/content/examples/http/src/app/heroes/heroes.component.css [hunk 2]
```diff
@@ -1,73 +1,66 @@
 /* HeroesComponent's private CSS styles */
+.search input {
+  width: 100%;
+  padding: .75rem;
+  margin: 1rem 0;
+  box-sizing: border-box;
+}
```
Comment (gkalpak) [inline aio/content/examples/http/src/app/heroes/heroes.component.css:7]:
```suggestion
}

```

File: aio/content/guide/http.md [hunk 3]
```diff
@@ -371,7 +371,7 @@ The following example shows how you can pipe a failed request to the `retry()` o
 
 In addition to fetching data from a server, `HttpClient` supports other HTTP methods such as PUT, POST, and DELETE, which you can use to modify the remote data.
 
-The sample app for this guide includes a simplified version of the "Tour of Heroes" example
+The sample app for this guide includes a abridged version of the "Tour of Heroes" example
```
Comment (gkalpak) [inline aio/content/guide/http.md:374]:
```suggestion
The sample app for this guide includes an abridged version of the "Tour of Heroes" example
```

File: aio/content/guide/http.md [hunk 4]
```diff
@@ -563,7 +563,7 @@ for each `HttpClient` method call.
 
 To implement an interceptor, declare a class that implements the `intercept()` method of the `HttpInterceptor` interface.
 
- Here is a do-nothing _noop_ interceptor that simply passes the request through without touching it:
+ Here is a do-nothing _noop_ interceptor that passes the request through without touching it:
```
Comment (gkalpak) [inline aio/content/guide/http.md:566]:
```suggestion
Here is a do-nothing _noop_ interceptor that passes the request through without touching it:
```

File: aio/content/examples/http/src/app/downloader/downloader.component.html [hunk 5]
```diff
@@ -1,4 +1,5 @@
-<h3>Download the textfile</h3>
+<hr />
```
Comment (gkalpak) [inline aio/content/examples/http/src/app/downloader/downloader.component.html:1]: I don't think it is a good idea to include this -context-specific styling elements in the component's templates. The horizontal separators should be added in [app.component.html#L19-L24][1]. (Same for all components used there.)

[1]: https://github.com/angular/angular/blob/019c3c211e449b6db08be463de4b9fea5b826376/aio/content/examples/http/src/app/app.component.html#L19-L24

**Review by gkalpak — CHANGES_REQUESTED** [review]
Awesome work :tada:

While you're at it, could you also add some margin (and possibly padding) to the buttons in [ConfigComponent][1] and [DownloaderComponent][2]?

[1]: https://github.com/angular/angular/blob/019c3c211e449b6db08be463de4b9fea5b826376/aio/content/examples/http/src/app/config/config.component.html#L3-L6
[2]: https://github.com/angular/angular/blob/019c3c211e449b6db08be463de4b9fea5b826376/aio/content/examples/http/src/app/downloader/downloader.component.html#L3-L4

File: aio/content/examples/http/src/app/heroes/heroes.component.css [hunk 6]
```diff
@@ -1,88 +1,56 @@
...
-  margin-bottom: 2px;
-  width: 11em;
-}
-
 .heroes input {
-  position: relative;
-  top: -3px;
-  width: 12em;
+  max-width: 12rem;
 }
```
Comment (gkalpak) [inline aio/content/examples/http/src/app/heroes/heroes.component.css:56]: Not sure which styles need to change, but the `.heroes` styles look off when editing a hero:

edit-hero

**Review by gkalpak — CHANGES_REQUESTED** [review]
A few minor comments. Otherwise lgtm 🚀
Great work, @kapunahelewong :100:

File: aio/content/examples/http/src/app/messages/messages.component.html [hunk 7]
```diff
@@ -1,5 +1,5 @@
+<h2>Messages</h2>
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/examples/http/src/app/messages/messages.component.html:1]: Ok. moving back in. I found it jarring for the whole section to disappear but I can go either way.

File: aio/content/examples/http/src/app/config/config.component.ts [hunk 8]
```diff
@@ -7,7 +7,7 @@ import { Config, ConfigService } from './config.service';
   selector: 'app-config',
   templateUrl: './config.component.html',
   providers: [ ConfigService ],
-  styles: ['.error {color: red;}']
+  styles: ['.error {color: #b30000;}']
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/examples/http/src/app/config/config.component.ts:10]: Thanks for showing me the right way to spacing these. I'm never quite certain when putting styles in here. The color `red` on a white bg failed WebAIM's contrast checker, so I darkened it until it passed.

File: aio/content/examples/http/src/app/heroes/heroes.component.css (hunk 6 above)
Comment (kapunahelewong) [reply aio/content/examples/http/src/app/heroes/heroes.component.css:56]: Oh thanks! Fixed.
Screen Shot 2021-03-25 at 10 32 21 AM

File: aio/content/examples/http/src/app/messages/messages.component.html [hunk 9]
```diff
@@ -1,5 +1,6 @@
+
```
Comment (gkalpak) [inline aio/content/examples/http/src/app/messages/messages.component.html:1]:
```suggestion
```

**Review by gkalpak — APPROVED** [review]
🚀

**Review by jelbourn — APPROVED** [review]
lgtm

**Review by aikithoughts — APPROVED** [review]
Reviewed-for: global-docs-approvers

**Pull request closed without merging**